core-bundle/contao/dca/tl_page.php

```diff
@@ -68,9 +68,12 @@
 			'keys' => array
 			(
 				'id' => 'primary',
+				'tstamp' => 'index',
+				'sorting' => 'index',
 				'alias' => 'index',
 				'type,dns' => 'index',
-				'pid,published,type,start,stop' => 'index'
+				'pid,published,type,start,stop' => 'index',
+				'dns,fallback,published,start,stop' => 'index',
```

Comment:
Instead of adding this index I think I’d try to add a `tl_page.type='root'` to `PageModel::findPublishedFallbackByHostname()`. That should fix the query speed as it can use the existing `type,dns` index just fine.
Additionally we could change `type,dns` to `type,dns,fallback,published`.

Reply:
But now the index is `type,dns,fallback,published,start,stop` instead of just `type,dns,fallback,published`?

Reply:
Yes, because that's even better if you use `start` and `stop` on top which we do for published checks. And if you do not query for `start` and `stop` then the index is still valid.

Reply:
I found the following `WHERE` combinations in the code base:

* `type`, `dns`, `published`, `start`, `stop`
* `type`, `fallback`, `dns`
* `type`, `published`, `start`, `stop`

Are all these variants covered by the `type,dns,fallback,published,start,stop` index, even though `fallback` and `published` are never in the same query?

Reply:
No of course not, all of them have to be covered or adjusted. Maybe we should add comments to the indexes like so:

```php
'dns,fallback,published,start,stop' => 'index', // Used by PageModel::findPublishedFallbackByHostname()
```

What do you think?

Also, please add a suggestion for the other indexes then.

Reply:
1. is not needed I think as the existing `pid,published,…` index already reduces the result to a very small number of rows
2. this index already exists as `pid,published,type,start,stop`
3. same as 1., there should we very few rows with the same alias, so the `alias` index should be enough here
4. already exists in this PR
5. same as 1. the existing `type,dns,…` index would alreay result in very few rows.

So I don’t think we need to add something here for the mentioned queriey